pkg/route/api/types.go

```diff
@@ -42,6 +42,10 @@ type RouteSpec struct {
 
 	//TLS provides the ability to configure certificates and termination for the route
 	TLS *TLSConfig
+
+	// Wildcard policy if any for the route.
+	// Currently only 'Subdomain' or 'None' is allowed.
+	WildcardPolicy WildcardPolicyType
```

Comment:
I would also expect wildcardpolicy on `RouteIngress` so clients can know what wildcard policy the router is using. @smarterclayton, that match what you would expect?